tsuru/client/init_test.go

```diff
@@ -0,0 +1,161 @@
+			createdFiles = append(createdFiles, c.Name())
+		}
+	}
+	if len(createdFiles) != 3 {
+		err = errors.New("Tsuru init failed to create a file")
+	}
+	c.Assert(err, check.IsNil)
+	err = tpath.Close()
+	c.Assert(err, check.IsNil)
+	err = os.RemoveAll(fkdir3)
+	c.Assert(err, check.IsNil)
+}
+
+func (s *S) TestInitInfo(c *check.C) {
+	c.Assert((&Init{}).Info(), check.NotNil)
+}
+
+func (s *S) TestCopyGitIgnoreWithGitIgnore(c *check.C) {
+	expected := "vendor/\n.git\n.gitignore\n"
+	wd, err := os.Getwd()
+	c.Assert(err, check.IsNil)
+	defer os.Chdir(wd)
+	testPath := filepath.Join("testdata", "deploy3")
+	err = os.Chdir(filepath.Join(wd, testPath))
+	c.Assert(err, check.IsNil)
+	err = copyGitIgnore()
+	c.Assert(err, check.IsNil)
+	defer ioutil.WriteFile(".tsuruignore", []byte(""), 0644)
+	tsuruignore, err := ioutil.ReadFile(".tsuruignore")
+	c.Assert(err, check.IsNil)
+	c.Assert(string(tsuruignore), check.Equals, expected)
+}
+
+func (s *S) TestCopyGitIgnoreWithoutGitIgnore(c *check.C) {
+	expected := ".git\n.gitignore\n"
+	wd, err := os.Getwd()
+	c.Assert(err, check.IsNil)
+	defer os.Chdir(wd)
+	tmpDir := os.TempDir()
+	err = os.Chdir(tmpDir)
+	c.Assert(err, check.IsNil)
+	err = copyGitIgnore()
+	c.Assert(err, check.IsNil)
+	tsuruignore, err := ioutil.ReadFile(".tsuruignore")
+	c.Assert(err, check.IsNil)
+	c.Assert(string(tsuruignore), check.Equals, expected)
+	err = os.Remove(filepath.Join(tmpDir, ".tsuruignore"))
+	c.Assert(err, check.IsNil)
+}
+
+func (s *S) TestWriteTsuruYaml(c *check.C) {
+	deploy3 := filepath.Join("testdata", "deploy3")
+	wd, err := os.Getwd()
+	c.Assert(err, check.IsNil)
+	tPath := filepath.Join(wd, deploy3, "fakeDir")
+	err = os.Chdir(filepath.Join(wd, deploy3))
+	c.Assert(err, check.IsNil)
+	err = os.Mkdir("fakeDir", os.ModePerm)
+	c.Assert(err, check.IsNil)
+	defer os.Chdir(wd)
+	err = os.Chdir("fakeDir")
+	c.Assert(err, check.IsNil)
+	_, err = os.Create("tsuru.yaml")
+	c.Assert(err, check.IsNil)
+	err = writeTsuruYaml()
+	c.Assert(err, check.IsNil)
+	data, err := ioutil.ReadFile("tsuru.yaml")
+	c.Assert(err, check.IsNil)
+	c.Assert(data, check.NotNil)
+	err = os.RemoveAll(tPath)
+	c.Assert(err, check.IsNil)
+}
+
+func (s *S) TestWriteProcfile(c *check.C) {
+	deploy3 := filepath.Join("testdata", "deploy3")
+	wd, err := os.Getwd()
+	c.Assert(err, check.IsNil)
+	tPath := filepath.Join(wd, deploy3, "fakeDirP")
+	err = os.Mkdir(tPath, os.ModePerm)
+	c.Assert(err, check.IsNil)
+	defer os.Remove(tPath)
+	err = os.Chdir(tPath)
+	c.Assert(err, check.IsNil)
+	defer os.Chdir(wd)
+	_, err = os.Create("Procfile")
+	c.Assert(err, check.IsNil)
+	err = writeProcfile()
+	c.Assert(err, check.IsNil)
+	data, err := ioutil.ReadFile("Procfile")
+	c.Assert(err, check.IsNil)
+	c.Assert(data, check.NotNil)
+	err = os.Chdir(filepath.Join(wd, "testdata", "deploy3"))
+	c.Assert(err, check.IsNil)
+	err = os.Remove(filepath.Join(tPath, "Procfile"))
+	c.Assert(err, check.IsNil)
+}
+
+func (s *S) TestInitRun(c *check.C) {
```

Comment:
Instead of testing each function separately, I think you could test all cases from `Init` command, which is the public interface. What do you think?